### DISCUSSION TITLE
Possible to mount another application?

### DISCUSSION THREAD

**Discussion by jonathan-s** [sparckles/Robyn#748] [Q&A] [answered]
Both flask and fastapi has the option of mounting another application, or being mounted. See [fastapi](https://fastapi.tiangolo.com/advanced/sub-applications/) and [flask](https://flask.palletsprojects.com/en/2.3.x/patterns/appdispatch/) docs. 

This would be hugely advantageous if you're deciding to migrate from one application framework to robyn for instance, as you can make the transition smoothly and not rewrite the entire application.

**Comment by hasansezertasan** [comment]
I have a project that gathers examples to mount different framework applications into each other.

Here is the project [hasansezertasan/a2wsgi-examples](https://github.com/hasansezertasan/a2wsgi-examples).

Sadly, I have failed to mount Robyn application to a FastAPI application.

**Reply by jonathan-s** [reply]
Awesome resource! Thanks for your work there!

**Reply by hasansezertasan** [reply]
You're welcome. I just checked the entire repository. There is no talking about ASGI and WSGI, I believe this package doesn't sport ASGI/WSGI, so a2wsgi won't work here.

See #366, I believe they meant to add support for ASGI in last year but I couldn't see any sign of that.

#27, #125, #267, #268, and #312

**Comment by sansyrox** [comment, marked as answer]
Hey @jonathan-s 👋 

https://robyn.tech/documentation/example_app/subrouters_and_views#subrouter-and-views , you can mount sub Robyn apps like this.

However, it is not possible to mount a Robyn app in another web framework(at least to my knowledge) .


And addressing @hasansezertasan concerns regarding ASGI support. While the support is in plans, I am not actively working on it due to other higher prior items. 

I am more than happy to accept/review PRs regarding the same.

I hope this helps 😄 

Feel free to reopen the discussion if you have any more questions!

**Reply by jonathan-s** [reply]
Thank you! Though semantically mounting routes is a bit different than mounting apps, even if it's just within Robyn. The subrouter is the equivalent of blueprints in flask or routers in fastapi afterall.
